SYCL/Basic/code_location_e2e.cpp

```diff
@@ -0,0 +1,142 @@
+// RUN: %clangxx -fsycl -DNDEBUG %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out %CPU_CHECK_PLACEHOLDER
+// RUN: %GPU_RUN_PLACEHOLDER %t.out %GPU_CHECK_PLACEHOLDER
+// RUN: %ACC_RUN_PLACEHOLDER %t.out %ACC_CHECK_PLACEHOLDER
+
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out %CPU_CHECK_PLACEHOLDER
+// RUN: %GPU_RUN_PLACEHOLDER %t.out %GPU_CHECK_PLACEHOLDER
+// RUN: %ACC_RUN_PLACEHOLDER %t.out %ACC_CHECK_PLACEHOLDER
+
+/*
+    clang++ -fsycl -DNDEBUG -o smyl.bin code_location_e2e.cpp   // <<--- NDEBUG
+   will suppress fileLocation
+
+    -DNDEBUG  <-- not debugging, meaning fileName should not be retrievable (for
+   data privacy reasons)
+
+    This test uses hardcoded file code location positions. If modified, those
+   may change. pass in any value to signal "report only" and not run tests.
+
+*/
+
+// clang-format off
+//   we turn off clang-format, so that the line numbers don't get changed by reformatting.
+
+#include <sycl/sycl.hpp>
+using namespace sycl;
+
+bool stringsAreSameP(const char *a, const char *b){
+    // ensure a is not null
+    if(a == nullptr){
+        if(b==nullptr){ return true; } // hey, they match after all!
+
+        return false;
+    }
```

Comment:
Please don't disable `clang-format`. This looks so artificial...

Reply:
it's only a test. because this test uses hard coded code locations, it's best to keep clang-format off, otherwise people that have to edit this in the future may run into difficulties.

Reply:
I cannot agree with this. Current violation of the formatting/style in this file speak for themselves.